docs/source/getting-started/testing.rst

```diff
@@ -0,0 +1,105 @@
+.. Copyright (c) 2020 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Testing Your App
+****************
+
+When developing your application, you will want to test certain workflows through the UI.
+In particular, you'll want to make sure that changes such as new features do not break existing functionality by mistake.
+It may be fine to do this testing manually at first, but it quickly becomes more tiresome and error prone as the app grows.
+
+In this section we will show you one approach to automatically testing workflows in your UI.
+This will allow you to iterate on your app faster and with more confidence that you aren't breaking it!
+
+Testing with Jest and Puppeteer
+===============================
+
+There are two tools that we chose to write end to end tests for our app.
+Of course there are more to choose from but we will show you one path here.
+The first tool is a general JavaScript testing framework called
+`Jest <https://jestjs.io/>`_.
+Jest allows you to write test cases containing expectations about behaviour should occur.
+If the expectations for a test are not met, then you should see helpful messages about what went wrong.
+You can also write setup and cleanup procedures that run before and after each test.
+This will be very useful for us to manage services like the DAML sandbox and JSON API server.
+
+The second tool we use heavily is a library called
+`Puppeteer <https://pptr.dev/>`_.
+Puppeteer allows you to control a Chrome browser from a NodeJS program.
+In our case we use it to perform actions like logging in, following friends and sending messages.
+We can also control multiple browser pages at once and test the interactions between different users.
+As a bonus, if something goes wrong you can take a screenshot and see what state your test browser got into!
+
+Let's see how to use these tools to write some tests for our social network app.
+You can see the full suite in the file ``index.test.tsx``.
+We first have some global state that we will use throughout our tests.
+Specifically, we have child processes for the ``daml start`` and ``yarn start`` commands, which run for the duration of our tests.
+We also have a single Puppeteer browser that we share among tests, opening new pages in each.
+
```

Comment:
Would be great to have a new subsection here "Initialising tests" or something like that. In general adding more structure.